compiler/noirc_frontend/src/tests/metaprogramming.rs

```diff
@@ -183,6 +187,117 @@ fn generate_function_with_macros() {
     ");
 }
 
+#[test]
+fn generate_function_with_macros_on_trait() {
+    let src = "
+    #[foo]
+    trait MyTrait {}
+
+    comptime fn foo(_f: TraitDefinition) -> Quoted {
+        quote {
+            pub fn bar(x: i32) -> i32  {  
+                let y = x + 1;
+                y + 2
+            }
+        }
+    }
+
+    struct Foo {}
```

Comment:
I'm guessing that you implement `MyTrait` for `Foo` so that `MyTrait` is used and then write `let _ = Foo {}` in `main` so that `Foo` is used? If so, why not use `impl MyTrait for ()` instead?